sysapi/include/sysapi_util.h

```diff
@@ -117,52 +124,40 @@ typedef struct {
                             // parsing sessions following handles section.
 } COMMAND_HANDLES;
 
-struct TSS2_SYS_CONTEXT;
-
-TPM_RC CopyCommandHeader(TSS2_SYS_CONTEXT *sysContext, TPM_CC commandCode);
-UINT16 GetDigestSize( TPM_ALG_ID authHash );
-UINT32 GetCommandSize( TSS2_SYS_CONTEXT *sysContext );
-
-TPM_RC ConcatSizedByteBuffer( TPM2B_MAX_BUFFER *result, TPM2B *addBuffer );
-
-void InitSysContextFields( TSS2_SYS_CONTEXT *sysContext );
-void InitSysContextPtrs ( TSS2_SYS_CONTEXT *sysContext, size_t contextSize );
+TPM_RC CopyCommandHeader(_TSS2_SYS_CONTEXT_BLOB *ctx, TPM_CC commandCode);
+UINT16 GetDigestSize(TPM_ALG_ID authHash);
+UINT32 GetCommandSize(_TSS2_SYS_CONTEXT_BLOB *ctx);
 
-TSS2_RC CompleteChecks( TSS2_SYS_CONTEXT *sysContext );
-
-TSS2_RC CommonComplete( TSS2_SYS_CONTEXT *sysContext );
+TPM_RC ConcatSizedByteBuffer(TPM2B_MAX_BUFFER *result, TPM2B *addBuffer);
+void InitSysContextFields(_TSS2_SYS_CONTEXT_BLOB *ctx);
+void InitSysContextPtrs(_TSS2_SYS_CONTEXT_BLOB *ctx, size_t contextSize);
+TSS2_RC CompleteChecks(_TSS2_SYS_CONTEXT_BLOB *ctx);
+TSS2_RC CommonComplete(_TSS2_SYS_CONTEXT_BLOB *ctx);
 
 TSS2_RC  CommonOneCallForNoResponseCmds(
-    TSS2_SYS_CONTEXT *sysContext,
+    _TSS2_SYS_CONTEXT_BLOB *ctx,
```

Comment:
Is it possible to flipp the paradigm here and left everything coming into the routines as a TSS2_SYS_CONTEXT pointer and casted them to something else, perhaps a TSS2_INTERNAL_SYS_CONTEXT pointer and just renamed the structures?
 I'm not sure if this is possible due to some API breakage.

Reply:
All the external Tss2_Sys_*() functions take TSS2_SYS_CONTEXT * opaque as per the spec.
We cast it too the _TSS2_SYS_CONTEXT_BLOB *, which is used internally.
We should not expose the internal/private _TSS2_SYS_CONTEXT_BLOB.
Is this what you had in mind?

Reply:
I think it boils down to the use of _ prefix to denote the internal cast of the opaque structure. But that appears to already be in place.